blog/2023-06-16-inline-sdk.mdx

```diff
@@ -0,0 +1,1104 @@
+
+        const currentRangeLength = end - start;
+        const percentageAlongLine =
+            (measure - start) / currentRangeLength;
+
+        // Move the video progress to the corresponding percentage through the video
+        // that matches the percentage along the line.
+        seekTo(percentageAlongLine * 100);
+
+        // Pause the video to be able to see where the position that was jumped to is.
+        if (playerState.isPlaying) {
+            togglePlay();
+        }
+    };
+    // highlight-end
+
+    return(...)
+}
+```
+
+The last thing we want to do is add a debounce to the handler to prevent any performance issues that can happen when moving around a lot. Install Lodash's debounce tool with
+
+```
+npm install lodash.debounce
+```
+
+Wrap the `handleCrosshairPositionChangeEvent` in a debounce higher order function. Give the debounce a 100ms refresh argument.
+
+```tsx title="src/bands/VideoBand/Band.tsx"
+// highlight-next-line
+import debounce from "lodash.debounce";
+...
+export const Band: React.FC<BandProps> = (props) => {
+    ...
+
+    // highlight-start
+    const handleCrosshairPositionChangeEvent = debounce((arg: StationInfoEventArg) => {
+        ...
+    }, [100])
+    // highlight-end
+
+    return(...)
+}
+```
+
+With the event handler and the event listeners added, the `Band.tsx` file should look like this:
+
+```tsx title="src/bands/VideoBand/Band.tsx"
+import debounce from "lodash.debounce";
+import {
+    BiPlay,
+    BiPause,
+    BiVolumeFull,
+    BiVolumeMute,
+} from "react-icons/bi";
+import React, { useEffect, useRef } from "react";
+import useVideoPlayer from "./useVideoPlayer";
+import "./Band.css";
+import { BandProps, StationInfoEventArg } from "inline";
+
+export const Band: React.FC<BandProps> = (props) => {
+    const player = useRef<HTMLVideoElement>();
+    const {
+        playerState,
+        togglePlay,
+        handleOnTimeUpdate,
+        handleVideoSpeed,
+        toggleMute,
+        seekTo,
+    } = useVideoPlayer(player);
+
+    /**
+     * Subscribing to inline events can allow for better integration of custom bands into inline.
+     */
+    useEffect(() => {
+        const mapCrosshairEvent = props.InlineManager.event(
+            "ChartCrosshairLocationUpdateEvent"
+        ).subscribe(undefined, handleCrosshairPositionChangeEvent);
+        const chartCrosshairEvent = props.InlineManager.event(
+            "XValueChangeEvent"
+        ).subscribe(undefined, handleCrosshairPositionChangeEvent);
+        return () => {
+            props.InlineManager.event(
+                "ChartCrosshairLocationUpdateEvent"
+            ).unsubscribe(mapCrosshairEvent);
+            props.InlineManager.event(
+                "XValueChangeEvent"
+            ).unsubscribe(chartCrosshairEvent);
+        };
+    });
+
+    /**
+     * When the inline cursor moves, it should update the video progress to match.
+     */
+    const handleCrosshairPositionChangeEvent = debounce(
```

Comment:
This needed to be typed in the presentation.

Reply:
Fixed